pysages/_nn/objectives.py

```diff
@@ -0,0 +1,283 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2020-2021: PySAGES contributors
+# See LICENSE.md and CONTRIBUTORS.md at https://github.com/SSAGESLabs/PySAGES
+
+from jax import value_and_grad, vmap
+from plum import dispatch
+from typing import NamedTuple
+
+from .utils import (
+    Float,
+    number_of_weights,
+    pack,
+    prod,
+    sum_squares,
+    unpack,
+)
+
+import jax.numpy as np
+import numpy as onp
+
+
+# Losses
```

Comment:
Are these all abstract classes, or why are they all defined here, but not implemented.
I am a little confused here.